modules/http/scanner.go

```diff
@@ -239,18 +239,33 @@ func (scan *scan) dialContext(ctx context.Context, network string, addr string)
 
 // getTLSDialer returns a Dial function that connects using the
 // zgrab2.GetTLSConnection()
-func (scan *scan) getTLSDialer(t *zgrab2.ScanTarget) func(net, addr string) (net.Conn, error) {
-	return func(net, addr string) (net.Conn, error) {
-		outer, err := scan.dialContext(context.Background(), net, addr)
+func (scan *scan) getTLSDialer(t *zgrab2.ScanTarget) func(network, addr string) (net.Conn, error) {
+	return func(network, addr string) (net.Conn, error) {
+		outer, err := scan.dialContext(context.Background(), network, addr)
 		if err != nil {
 			return nil, err
 		}
-
 		cfg, err := scan.scanner.config.TLSFlags.GetTLSConfigForTarget(t)
 		if err != nil {
 			return nil, err
 		}
 
+		// Set SNI server name on redirects unless --server-name was used (issue #300)
+		//  - t.Domain is always set to the *original* Host so it's not useful for setting SNI
+		//  - host is the current target of the request in this context; this is true for the
+		//    initial request as well as subsequent requests caused by redirects
+		//  - scan.scanner.config.ServerName is the value from --server-name if one was specified
+
+		// If SNI is enabled and --server-name is not set, use the target host for the SNI server name
+		if !scan.scanner.config.NoSNI && scan.scanner.config.ServerName == "" {
+			host, _, err := net.SplitHostPort(addr)
+			if err != nil {
+				log.Errorf("http/scanner.go getTLSDialer: unable to split host:port '%s'", addr)
+				log.Errorf("Something went wrong splitting host/port: %s", err)
```

Comment:
Can you condense this down into one log message?

Reply:
Absolutely, meant to remove that, nice catch. Should be fixed now

Reply:
BTW, I think I also need to add logic to check if `host` is a an IP address. Currently, this will send an SNI server name value of the dotted quad (if a dotted quad is what is being used for the `Host:` header)

From [RFC4366](https://tools.ietf.org/html/rfc4366):

> Literal IPv4 and IPv6 addresses are not permitted in "HostName".

I think I need to add a check to see if it's an IP address and if it is, not set anything at all. Based on a quick test with this build and looking at a wire capture, I confirmed that with a build of this PR it is in fact setting the server name to a dotted quad when an IP only is specified

I think the answer is obviously, just checking in case you have some specific knowledge about how SNI works today and whether a deviation from the RFC would be a normal thing